api/service.ts

```diff
@@ -236,14 +236,14 @@ export interface HostPathVolumeConstructor {
 export let HostPathVolume: HostPathVolumeConstructor; // tslint:disable-line
 
 /**
- * The arguments to construct a Service object.
+ * The arguments to construct a Service object. These arguments may include container information, for simple
+ * single-container scenarios, or you may specify that information using the containers property.
  */
-export interface ServiceArguments {
+export interface ServiceArguments extends Container {
```

Comment:
I'll admit I don't love this API design.  It won't be clear what properties are required on the Service vs. on the Container.  For example, is `memory` needed on the service or on the container?  It will look like it can be provided either place, but will silently do nothing if passed on the service for the case where `containers` is specified.

Reply:
If we do want to go this direction though, I wonder if we should instead use a union type of `ServiceArguments` and `SingleContainerServiceArguments` so that we can be clear that the former doesn't allow container-level properties on the service?

Reply:
I am fine either way, all I care about is the usage code being beautiful.

Just lmk what you'd like me to do here, but I'd like to land soon so I can get this blog post out.

Reply:
(I also think a fine answer would be to use service-wide settings as defaults if not specified explicitly in the containers map)

Reply:
Let's go with your current change here for now.  We can add the better TS typing on top of this later.  

> (I also think a fine answer would be to use service-wide settings as defaults if not specified explicitly in the containers map)

Unfortunately I don't think that can work - the meaning is different as a service-level setting vs. as a per-container setting in a multi-container service.